src/campaigns/tasks/services/aiCampaignManager.service.ts

```diff
@@ -0,0 +1,214 @@
+import { HttpService } from '@nestjs/axios'
+import { BadGatewayException, Injectable, Logger } from '@nestjs/common'
+import { lastValueFrom } from 'rxjs'
+import { Methods } from 'http-constants-ts'
+import { AxiosResponse } from 'axios'
+
+export interface StartCampaignPlanRequest {
+  candidate_name: string
+  election_date: string
+  office_and_jurisdiction: string
+  race_type: string
+  incumbent_status: string
+  seats_available: number
+  number_of_opponents: number
+  win_number: number
+  total_likely_voters: number
+  available_cell_phones: number
+  available_landlines: number
+  primary_date?: string | null
+  additional_race_context?: string | null
+}
+
+export interface CampaignPlanSession {
+  session_id: string
+}
+
+export interface ProgressStreamData {
+  progress: number
+  status: 'processing' | 'completed' | 'failed'
+  message: string
+  logs: string[]
+  timestamp: string
+  has_pdf: boolean
+  has_json: boolean
+  download_links: {
+    pdf?: string
+    json?: string
+  }
+  expires_at: string | null
+  expires_at_formatted: string | null
+  files_ready: {
+    pdf: boolean
+    json: boolean
+    total: number
+  }
+}
+
+@Injectable()
+export class AiCampaignManagerService {
+  private readonly apiBaseUrl: string | undefined
+  private readonly logger = new Logger(AiCampaignManagerService.name)
+
+  constructor(private readonly httpService: HttpService) {
+    this.apiBaseUrl = process.env.AI_CAMPAIGN_MANAGER_BASE
+  }
+
+  async startCampaignPlanGeneration(
+    request: StartCampaignPlanRequest,
+  ): Promise<CampaignPlanSession> {
+    try {
+      const response = await this.fetchFromApi<CampaignPlanSession>(
+        '/start-campaign-plan-generation',
+        {
+          method: Methods.POST,
+          data: request,
+        },
+      )
+
+      return response.data
+    } catch (error) {
+      this.logger.error('Failed to start campaign plan generation', error)
+      throw new BadGatewayException('Failed to start campaign plan generation')
+    }
+  }
+
+  async downloadJson(sessionId: string): Promise<unknown> {
```

Comment:
Can we not type this `unknown`? Do we not know what the response structure from the other service is?